create.go

```diff
@@ -0,0 +1,322 @@
+					os.Exit(-1)
+				}
+				args = append(args, "--"+goption, abs_path)
+			}
+		}
+		args = append(args,
+			"containerd", "--solo-namespaced",
+			"--containerd-dir", namespace,
+			"--state-dir", root,
+			"--listen", filepath.Join(namespace, "namespaced.sock"),
+		)
+		cmd = &exec.Cmd{
+			Path: path,
+			Args: append([]string{"runv"}, args...),
+		}
+		cmd.Dir = "/"
+		cmd.SysProcAttr = &syscall.SysProcAttr{
+			Setsid: true,
+		}
+		err = cmd.Start()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to launch runv containerd: %v\n", err)
+			os.Exit(-1)
+		}
+		if _, err = os.Stat(filepath.Join(namespace, "namespaced.sock")); os.IsNotExist(err) {
+			time.Sleep(3 * time.Second)
+		}
+	}
+
+	address := filepath.Join(namespace, "namespaced.sock")
+	err = createContainer(context, container, address, spec)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error %v\n", err)
+		cmd.Process.Signal(syscall.SIGINT)
+		os.Exit(-1)
+	}
+	err = os.Symlink(namespace, filepath.Join(root, container, "namespace"))
+	if err != nil {
+		cmd.Process.Signal(syscall.SIGINT)
+		os.Exit(-1)
+	}
+	if !createOnly {
+		startContainer(context, bundle, container, address, spec, context.Bool("detach"))
+	}
+	os.Exit(0)
+}
+
+// Shared namespaces multiple containers suppurt
+// The runv supports pod-style shared namespaces currently.
+// (More fine grain shared namespaces style (docker/runc style) is under implementation)
+//
+// Pod-style shared namespaces:
+// * if two containers share at least one type of namespace, they share all kinds of namespaces except the mount namespace
+// * mount namespace can't be shared, each container has its own mount namespace
+//
+// Implementation detail:
+// * Shared namespaces is configured in Spec.Linux.Namespaces, the namespace Path should be existing container name.
+// * In runv, shared namespaces multiple containers are located in the same VM which is managed by a runv-daemon.
+// * Any running container can exit in any arbitrary order, the runv-daemon and the VM are existed until the last container of the VM is existed
+
+func createContainer(context *cli.Context, container, address string, config *specs.Spec) error {
+	pid := os.Getpid()
+	stdin, _ := os.Readlink(fmt.Sprintf("/proc/%d/fd/0", pid))
+	stdout, _ := os.Readlink(fmt.Sprintf("/proc/%d/fd/1", pid))
+	stderr, _ := os.Readlink(fmt.Sprintf("/proc/%d/fd/2", pid))
+	r := &types.CreateContainerRequest{
+		Id:         container,
+		Runtime:    "runv-create",
+		BundlePath: context.String("bundle"),
+		Stdin:      stdin,
+		Stdout:     stdout,
+		Stderr:     stderr,
+	}
+
+	c := getClient(address)
+	if _, err := c.CreateContainer(netcontext.Background(), r); err != nil {
+		return err
+	}
+	if context.String("pid-file") != "" {
+		stateData, err := ioutil.ReadFile(filepath.Join(context.GlobalString("root"), container, stateJson))
+		if err != nil {
+			return err
+		}
+
+		var s specs.State
+		if err := json.Unmarshal(stateData, &s); err != nil {
+			return err
+		}
```

Comment:
You can move 
```
+		stateData, err := ioutil.ReadFile(filepath.Join(context.GlobalString("root"), container, stateJson))
 +		if err != nil {
 +			return err
 +		}
 +
 +		var s specs.State
 +		if err := json.Unmarshal(stateData, &s); err != nil {
 +			return err
 +		}
```

out of the `if`. Because we need to create `state.json` no matter `--pid-file` is set or not.